### PR TITLE
Better hostname handling

### PR DESCRIPTION
**Pull request by jumaffre**
Further to https://github.com/microsoft/CCF/issues/730, `cchost` now sanitises the `rpc-address` host to make sure that if it does not specify an IP address, then `--domain` should be set. Also tweaked the docs slightly.

### CODE REVIEW

File: src/host/main.cpp [hunk 1]
```diff
@@ -274,6 +275,14 @@ int main(int argc, char** argv)
     public_rpc_address = rpc_address;
   }
 
+  if (domain.empty() && !ds::is_valid_ip(rpc_address.hostname.c_str()))
+  {
+    throw std::logic_error(fmt::format(
```
Comment (achamayou) [inline src/host/main.cpp]: What happens with "localhost"?

Comment (jumaffre) [reply src/host/main.cpp]: As I expected: same as any other non-IP hostname.

**Comment by cimetrics** [comment]
better_hostname@[4456 aka 20200131.7](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=4456&view=results) vs master ewma over 30 builds from [4022](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=4022&view=results) to [4447](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=4447&view=results)
images

File: src/ds/net.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  inline std::optional<IPAddr> ip_to_binary(const char* hostname)
+  {
+    IPAddr ip_bin;
+    ip_bin.size = ipv4_binary_size;
+    if (inet_pton(AF_INET, hostname, ip_bin.buf) != 1)
+    {
+      ip_bin.size = ipv6_binary_size;
+      if (inet_pton(AF_INET6, hostname, ip_bin.buf) != 1)
+      {
+        return {};
```
Comment (eddyashton) [inline src/ds/net.h]: Please use `return std::nullopt` for clarity.

File: sphinx/source/operators/start_network.rst [hunk 3]
```diff
@@ -18,7 +18,8 @@ To create a new CCF network, the first node of the network should be started wit
     --enclave-type debug
     --node-address node_ip:node_port
     --rpc-address <ccf-node-address>
-    --public-rpc-address public_rpc_ip:public_rpc_port
+    --public-rpc-address <ccf-node-public-address>
+    [--domain domain]
```
Comment (iqsarv) [inline sphinx/source/operators/start_network.rst]: does it make sense to `s/domain/node-fqdn` or mention in the docs that this is node's FQDN to make more explicit?

i was initially confused with `domain` as network's domain, that can accept wild-card characters like `*.blockchian.azure.com`. 

p.s sorry for post-merge comments

**Pull request merged**